### PR TITLE
Add local storage

### PR DESCRIPTION
**Pull request by Boyko-Anna**

### CODE REVIEW

**Comment by Boyko-Anna** [comment]
The data are stored in local storage, and they expire every day at 12 am. Hope the logic in checkDataExpire function works as expected...
I need some suggestions how to improve my code, cause as for me it looks really awful.

File: index.html [hunk 1]
```diff
@@ -24,9 +24,9 @@
 
         <div class="forecastDetails">
             <div class="scale-container"> 
-                <input type="radio" name="toggleScale" onclick="toggleScale('M')"id="C" value="metric" class="input-hide" checked></input>
+                <input type="radio" name="toggleScale" onclick="toggleScale()" id="C" value="M" class="input-hide" checked></input>
```
Comment (viktor-yakubiv) [inline index.html]: Much better!

File: js/app.js [hunk 2]
```diff
@@ -1,43 +1,120 @@
...
         mainForecastItem.appendChild(details);
+
+        return true;
+
+    }
     });
 }
 
-function generateSummary (data) {
+function generateSummary (data, scale) {
```
Comment (viktor-yakubiv) [inline js/app.js]: Awesome!

File: js/app.js [hunk 3]
```diff
@@ -1,43 +1,120 @@
...
+function extractFromStorage(key) {
+    return localStorage.getItem(key);
+}
+
+function checkDataExpire(storageData) {
+    let storageTimeStamp = storageData.data[0].ts * 1000;
+    let storageDate = new Date(storageTimeStamp);
+    let storageHour = storageDate.getHours();
+    let storageExpires = storageTimeStamp + ((24 - storageHour) * 60 * 60 * 1000);
+    let currentTimeStamp = Date.now();
```
Comment (viktor-yakubiv) [inline js/app.js]: It can be muuuuch easier.

```js
storageTimestamp - Date.now() > cacheLiveTime
```

Comment (viktor-yakubiv) [reply js/app.js]: Even more, you can save in storage date kind `expiresAt` and check it like following:

```js
const cached = cache.get(query)

if (!cached || Date.now() > cached.expiresAt) {
  cache.delete(query)
  return api.call(query)
}

return cached
```

Comment (viktor-yakubiv) [reply js/app.js]: Even more, you may avoid deletion from cache by extending `Map`:

```js
const TIME_15MIN = 15 * 60 * 1000; // milliseconds

class Cache extends Map {
  get(key) {
    const value = super().get(key)

    if (value && value.expiresAt < Date.now()) {
      this.delete(key)
      return null
    }

    return value.payload
  }

  put(key, value, expiresAt = Date.now() + TIME_15MIN) {
    return super().put(key, { expiresAt, payload: value })
  }
}
```

**Pull request closed without merging**